packages/SwingSet/src/kernel/kernel.js

```diff
@@ -572,72 +682,114 @@ export default function buildKernel(kernelEndowments) {
       helpers,
       kernelKeeper,
       kernelKeeper.allocateVatKeeperIfNeeded(vatID),
+      vatPowers,
     );
   }
 
   // This kernel (and the genesis vats) will have been evaluated in a context in
   // which these canonical symbols were available. That was achieved by running
   // makeRequire() in controller.js. We want to provide the same context when we
   // evaluate dynamic vats.
```

Comment:
This comment applies mostly to the deleted code. Is there some aspect that applies also to vatRequire? Should be reworded to apply more directly to what's left.